app/modules/tmr.c

```diff
@@ -65,21 +65,54 @@ typedef void (*alarm_timer_callback)(void *arg);
 static alarm_timer_callback alarm_timer_cb[NUM_TMR] = {alarm_timer_cb0,alarm_timer_cb1,alarm_timer_cb2,alarm_timer_cb3,alarm_timer_cb4,alarm_timer_cb5,alarm_timer_cb6};
 
 // Lua: delay( us )
-static int tmr_delay( lua_State* L )
+static int tmr_delay_us( lua_State* L )
 {
   s32 us;
   us = luaL_checkinteger( L, 1 );
   if ( us <= 0 )
     return luaL_error( L, "wrong arg range" );
-  unsigned sec = (unsigned)us / 1000000;
-  unsigned remain = (unsigned)us % 1000000;
+
+     os_delay_us( us );
+     WRITE_PERI_REG(0x60000914, 0x73); //Watchdog clear to aviod system reboot
+
+  return 0;  
+}
+
+// Lua: delay( ms )
+static int tmr_delay_ms( lua_State* L )
+{
+  s32 ms;
+  ms = luaL_checkinteger( L, 1 );
+  if ( ms <= 0 )
+    return luaL_error( L, "wrong arg range" );
+    
   int i = 0;
-  for(i=0;i<sec;i++){
-    os_delay_us( 1000000 );
-    WRITE_PERI_REG(0x60000914, 0x73);
-  }
-  if(remain>0)
-    os_delay_us( remain );
+  for(i=0;i<1000;i++) // 1ms == 1000us. Simple stupid and should work.
```

Comment:
The main idea was to remove the original code with a lot of math. Just because of a huge latency for us operations. For example, if we are talking about ir protocols. So if I want to make a delay about 5us, I don't need any additional calculations. But I underdtand your idea about ms, it's good. I agree.